konfera/models/ticket.py

```diff
@@ -30,3 +32,14 @@ def __str__(self):
             first_name=self.first_name,
             last_name=self.last_name
         ).strip()
+
+    def save(self, *args, **kwargs):
+        if not hasattr(self, 'order'):
+            discount = 0
+            if self.discount_code:
+                discount = self.type.price * self.discount_code.discount/100
```

Comment:
Please create function discount_calculator() with logic above to calculate and return discount. We will most likely update this logic a bit more in near future.